include/fixed_containers/fixed_bitset.hpp

```diff
@@ -0,0 +1,536 @@
+
+    constexpr bool none() const noexcept {
+        return !any();
+    }
+
+    constexpr bool all() const noexcept {
+        constexpr bool _Zero_length = _Bits == 0;
+        if constexpr (_Zero_length) {  // must test for this, otherwise would count one full word
+            return true;
+        }
+
+        constexpr bool _No_padding = _Bits % _Bitsperword == 0;
+        for (size_t _Wpos = 0; _Wpos < _Words + _No_padding; ++_Wpos) {
+            if (_Data[_Wpos] != ~static_cast<_Ty>(0)) {
+                return false;
+            }
+        }
+
+        return _No_padding || _Data[_Words] == (static_cast<_Ty>(1) << (_Bits % _Bitsperword)) - 1;
+    }
+
+    constexpr FixedBitSet operator<<(size_t _Pos) const noexcept {
+        FixedBitSet _Tmp = *this;
+        _Tmp <<= _Pos;
+        return _Tmp;
+    }
+
+    constexpr FixedBitSet operator>>(size_t _Pos) const noexcept {
+        FixedBitSet _Tmp = *this;
+        _Tmp >>= _Pos;
+        return _Tmp;
+    }
+
+    constexpr _Ty _Getword(size_t _Wpos) const noexcept {  // nonstandard extension; get underlying word
+        return _Data[_Wpos];
+    }
+
+ private:
+    friend std::hash<FixedBitSet<_Bits>>;
+
+    static constexpr ptrdiff_t _Bitsperword = CHAR_BIT * sizeof(_Ty);
+    static constexpr ptrdiff_t _Words       = _Bits == 0 ? 0 : (_Bits - 1) / _Bitsperword;  // NB: number of words - 1
+
+    constexpr void _Trim() noexcept {  // clear any trailing bits in last word
+        constexpr bool _Work_to_do = _Bits == 0 || _Bits % _Bitsperword != 0;
+        if constexpr (_Work_to_do) {
+            _Data[_Words] &= (_Ty{1} << _Bits % _Bitsperword) - 1;
+        }
+    }
+
+    constexpr FixedBitSet& _Set_unchecked(size_t _Pos, bool _Val) noexcept {  // set bit at _Pos to _Val, no checking
+        auto& _Selected_word = _Data[_Pos / _Bitsperword];
+        const auto _Bit      = _Ty{1} << _Pos % _Bitsperword;
+        if (_Val) {
+            _Selected_word |= _Bit;
+        } else {
+            _Selected_word &= ~_Bit;
+        }
+
+        return *this;
+    }
+
+    constexpr FixedBitSet& _Flip_unchecked(size_t _Pos) noexcept {  // flip bit at _Pos, no checking
+        _Data[_Pos / _Bitsperword] ^= _Ty{1} << _Pos % _Bitsperword;
+        return *this;
+    }
+
+    [[noreturn]] constexpr void _Xinv(const std_transition::source_location& loc =
+        std_transition::source_location::current()) const {
+        CheckingType::invalid_argument("invalid FixedBitSet char", loc);
+    }
+
+    [[noreturn]] constexpr void _Xoflo(const std_transition::source_location& loc =
+        std_transition::source_location::current()) const {
+        CheckingType::std::overflow_error("FixedBitSet overflow", loc);
+    }
+
+    [[noreturn]] constexpr void _Xran(const std_transition::source_location& loc =
+        std_transition::source_location::current()) const {
+        CheckingType::std::out_of_range("invalid FixedBitSet position", loc);
+    }
+
+public:
+    _Ty _Data[_Words + 1];
```

Comment:
This has to be public to satisfy `IsStructuralType()` requirement. `std::array` has a similar quirk of making `__elems_[N]` public.